[Frontendv1] Make the execution time human readable on UI.

Alabhya268 opened the pull request:
fixes #3319
- [x] Make the execution time human readable on UI.

I have made execution time human readable.
I have not yet add ```day``` as I think it unecessay but let me know if its needed to be added.

![time](https://user-images.githubusercontent.com/57143358/111348726-79239400-86a6-11eb-9136-4b45d45a2854.png)

Alabhya268 commented:
@RishabhJain2018  @Ram81 , Please review this PR.

Ram81 commented on frontend/src/js/controllers/challengeCtrl.js:
```diff
@@ -872,6 +872,14 @@
+        vm.format_execution_time = function(execution_time) {
+            var execution_time_hr = Math.floor(execution_time/60/60);
+            var execution_time_min = Math.floor(execution_time/60) - (execution_time_hr * 60);
+            var execution_time_sec = Math.floor(execution_time % 60);
```
@Alabhya268 can we use date util to do the conversion. See [this](https://stackoverflow.com/questions/1322732/convert-seconds-to-hh-mm-ss-with-javascript) for reference

Alabhya268 replied:
@Ram81, I tried to do it in a way . It not working as expected.
Its adding the executions time in current time. I tried to modify it , I am able to get hour, min and sec but not days with this method.

Ram81 commented on frontend/src/js/controllers/challengeCtrl.js:
```diff
@@ -872,6 +872,14 @@
+            var execution_time_hr = Math.floor(execution_time/60/60);
+            var execution_time_min = Math.floor(execution_time/60) - (execution_time_hr * 60);
+            var execution_time_sec = Math.floor(execution_time % 60);
+            var formatted = execution_time_hr + 'hr ' + execution_time_min + 'min ' + execution_time_sec + 'sec';
```
Add space before `Hr` `Min` and `Sec`

Alabhya268 replied:
ok

RishabhJain2018 commented:
>I have not yet add day as I think it unecessay but let me know if its needed to be added.

Also, please do add day as some submissions run for over 36 hours.

Alabhya268 commented:
@Ram81, Please have a look at this PR, I have now done it with date util, but for some reason date was always 1 day ahead so I added ```-1``` , Could you please test it ?

@RishabhJain2018 , I have now added ```day``` option.
![time2](https://user-images.githubusercontent.com/57143358/111361977-4d0f0f80-86b4-11eb-9cc4-ea8acd699584.png)

RishabhJain2018 reviewed (changes requested):
@Alabhya268 Can we please create a filter like this: https://stackoverflow.com/a/30767487 which we can use at multiple places i.e. in all submissions and my submissions as well.

Alabhya268 commented:
> @Alabhya268 Can we please create a filter like this: https://stackoverflow.com/a/30767487 which we can use at multiple places i.e. in all submissions and my submissions as well.

I will try to do it .

Alabhya268 commented:
> @Alabhya268 Can we please create a filter like this: https://stackoverflow.com/a/30767487 which we can use at multiple places i.e. in all submissions and my submissions as well.

@RishabhJain2018 , I have made it as filter and it can now be used anywhere to convert seconds to datetime.

Alabhya268 commented:
> @Alabhya268 Can we please create a filter like this: https://stackoverflow.com/a/30767487 which we can use at multiple places i.e. in all submissions and my submissions as well.

@RishabhJain2018 , I have made it as filter and it can now be used anywhere to convert seconds to datetime.

RishabhJain2018 commented:
@Alabhya268 Can you please fix the Travis build?

Alabhya268 commented:
@RishabhJain2018 Travis build is now fixed.

Ram81 commented on frontend/src/js/filters/filters.js:
```diff
@@ -15,4 +15,21 @@
+            var hours = executiontime.getUTCHours();
+            var minutes = executiontime.getUTCMinutes();
+            var seconds = executiontime.getSeconds();
+            var timeString = days.toString().padStart(2, '0') + ' day ' + hours.toString().padStart(2, '0') + ' hr ' + 
```
@Alabhya268 let's not show `day` `'hr` or `min` when they have 0 value. Rest of the changes look fine.

cc: @RishabhJain2018

Alabhya268 commented:
@Ram81 , days , min , hours will now show accordingly.

![1](https://user-images.githubusercontent.com/57143358/111587851-c3ad2980-87e8-11eb-82f5-defe2027af29.png)
![2](https://user-images.githubusercontent.com/57143358/111587856-c576ed00-87e8-11eb-9b3d-7bbc229be4eb.png)
![3](https://user-images.githubusercontent.com/57143358/111587879-cc9dfb00-87e8-11eb-83da-f2282cdbeeb0.png)
![4](https://user-images.githubusercontent.com/57143358/111587882-cc9dfb00-87e8-11eb-9749-71fc599f3496.png)

Alabhya268 commented:
@Ram81 . Ping for PR

Alabhya268 commented:
@Ram81, Please review this PR again.

Ram81 commented:
@Alabhya268 If a submission takes 1 day we are showing 1 day and 00 sec we should either show all hour, mins and secs if we are showing day or we should only show when value is greater than 0.

![Screen Shot 2021-03-21 at 10 04 04 PM](https://user-images.githubusercontent.com/16323427/111912852-7d2e2800-8a91-11eb-81c6-a9f69ff8b88c.png)

@RishabhJain2018 what should we do? should we just stick to showing hh : mm : ss even when submission takes more than 24 hrs?

Alabhya268 commented:
> @Alabhya268 If a submission takes 1 day we are showing 1 day and 00 sec we should either show all hour, mins and secs if we are showing day or we should only show when value is greater than 0.
> 
> ![Screen Shot 2021-03-21 at 10 04 04 PM](https://user-images.githubusercontent.com/16323427/111912852-7d2e2800-8a91-11eb-81c6-a9f69ff8b88c.png)
> 
> @RishabhJain2018 what should we do? should we just stick to showing hh : mm : ss even when submission takes more than 24 hrs?

@Ram81 , I think we should stick full ``` hh : mm  : ss```  as UI would look odd if different challenges in column uses different format to show execution time. 

CC: @RishabhJain2018

Alabhya268 commented:
> @Alabhya268 If a submission takes 1 day we are showing 1 day and 00 sec we should either show all hour, mins and secs if we are showing day or we should only show when value is greater than 0.
> 
> ![Screen Shot 2021-03-21 at 10 04 04 PM](https://user-images.githubusercontent.com/16323427/111912852-7d2e2800-8a91-11eb-81c6-a9f69ff8b88c.png)
> 
> @RishabhJain2018 what should we do? should we just stick to showing hh : mm : ss even when submission takes more than 24 hrs?

@Ram81 @RishabhJain2018 , Which approch should I go for ?

Ram81 commented:
@Alabhya268 let's just show day hh mm sec when the value is > 0. And just show sec if submission time is > a minute. Let me know if you have any doubts

Alabhya268 commented:
> @Alabhya268 let's just show day hh mm sec when the value is > 0. And just show sec if submission time is > a minute. Let me know if you have any doubts

@Ram81 , It means we just need to show everthing till highest time right ?
like if execution is 1 hour , we'll show ```hh:mm:ss``` .

Alabhya268 commented:
> > @Alabhya268 let's just show day hh mm sec when the value is > 0. And just show sec if submission time is > a minute. Let me know if you have any doubts
> 
> @Ram81 , It means we just need to show everthing till highest time right ?
> like if execution is 1 hour , we'll show `hh:mm:ss` .

@Ram81 ?

Alabhya268 commented:
> > @Alabhya268 let's just show day hh mm sec when the value is > 0. And just show sec if submission time is > a minute. Let me know if you have any doubts
> 
> @Ram81 , It means we just need to show everthing till highest time right ?
> like if execution is 1 hour , we'll show `hh:mm:ss` .

@Ram81 , @RishabhJain2018

Alabhya268 commented:
> > @Alabhya268 let's just show day hh mm sec when the value is > 0. And just show sec if submission time is > a minute. Let me know if you have any doubts
> 
> @Ram81, you mean we show sec when time is < min right?
> 
> @Ram81, I am gonna make it so that `dd: hh: mm: ss` always show when execution time is > 0 second, This would make the execution time of the challenge list look even.

@RishabhJain2018, Please also state your thoughts.

Alabhya268 commented:
@Ram81, I've made it so that it will show ```dd: hh: mm: ss``` when execution time is > 0 and it will show ```00 sec``` when execution time is <1.

Ram81 commented:
@Alabhya268 can you share a screenshot with multiple submission each having different execution time. For ex: 10 sec, 1 min, 1 min 30 sec, 1 hour, 1 hour 20 min, 1 hour 30 sec, 1 hour 20 min 30 sec, 1 day, 1 day 10 sec, etc

Alabhya268 commented:
> @Alabhya268 can you share a screenshot with multiple submission each having different execution time. For ex: 10 sec, 1 min, 1 min 30 sec, 1 hour, 1 hour 20 min, 1 hour 30 sec, 1 hour 20 min 30 sec, 1 day, 1 day 10 sec, etc
@Ram81, Please look at these screenshot. If you have something different in mind then please let me know precisely what format should look.

![1hour 20 min](https://user-images.githubusercontent.com/57143358/112712081-0d85c600-8ef3-11eb-89ac-cab054be5245.png)
![1hour 30sec](https://user-images.githubusercontent.com/57143358/112712082-0e1e5c80-8ef3-11eb-8806-5cbd58a15bed.png)
![1min](https://user-images.githubusercontent.com/57143358/112712083-0e1e5c80-8ef3-11eb-8af0-67fe2bca6f42.png)
![1min 30 sec](https://user-images.githubusercontent.com/57143358/112712084-0eb6f300-8ef3-11eb-8fce-f3c0dc9bc102.png)
![10sec](https://user-images.githubusercontent.com/57143358/112712085-0f4f8980-8ef3-11eb-91eb-aa270d6eeb7e.png)
![s](https://user-images.githubusercontent.com/57143358/112712086-0f4f8980-8ef3-11eb-972e-8a0d7996fb0e.png)
![Screenshot from 2021-03-27 11-53-03](https://user-images.githubusercontent.com/57143358/112712087-0fe82000-8ef3-11eb-8768-b0a213512baa.png)
![1day](https://user-images.githubusercontent.com/57143358/112712075-0b236c00-8ef3-11eb-8f98-1a72d7dc0a81.png)
![1day 1 mon 12 sec](https://user-images.githubusercontent.com/57143358/112712076-0c549900-8ef3-11eb-80a4-d40a2df246a0.png)
![1day 10 sec](https://user-images.githubusercontent.com/57143358/112712077-0c549900-8ef3-11eb-9553-8cdf4b7306f6.png)
![1h 20 min 30 sec](https://user-images.githubusercontent.com/57143358/112712078-0ced2f80-8ef3-11eb-9f91-f18320a5a75a.png)
![1hour](https://user-images.githubusercontent.com/57143358/112712080-0ced2f80-8ef3-11eb-94b0-ddafecb05c0d.png)

Ram81 commented:
@Alabhya268 I think I didn't clearly mention it. We want to show `days`, `min`, `hr` only if the value is > 0. Here are the expected values for few cases:

24 hr 10 sec -> 01 day 10 secs
90 secs -> 1 min 30 secs
26 hr 2mins -> 1 day 122 mins

Let me know if you have any doubts

Alabhya268 commented:
> @Alabhya268 I think I didn't clearly mention it. We want to show `days`, `min`, `hr` only if the value is > 0. Here are the expected values for few cases:
> 
> 24 hr 10 sec -> 01 day 10 secs
> 90 secs -> 1 min 30 secs
> 26 hr 2mins -> 1 day 122 mins
> 
> Let me know if you have any doubts

@Ram81, I will incorporate the requested changes asap.

Alabhya268 commented:
> > @Alabhya268 I think I didn't clearly mention it. We want to show `days`, `min`, `hr` only if the value is > 0. Here are the expected values for few cases:
> > 24 hr 10 sec -> 01 day 10 secs
> > 90 secs -> 1 min 30 secs
> > 26 hr 2mins -> 1 day 122 mins
> > Let me know if you have any doubts
> 
> @Ram81, I will incorporate the requested changes asap.

@Ram81, Wouldn't we want to show 26 hours 2 min -> 1 day 2 hour 2 mins?

Ram81 commented:
Yes, updated the previous comment

Alabhya268 commented:
@Ram81, I've made execution time according to your requirements. Please have a look at the following images. 

![Screenshot from 2021-03-28 17-53-01](https://user-images.githubusercontent.com/57143358/112752562-486e2380-8ff1-11eb-96cf-a10fc8fc87e8.png)
![Screenshot from 2021-03-28 17-53-38](https://user-images.githubusercontent.com/57143358/112752563-499f5080-8ff1-11eb-8d48-7d58e9003ea4.png)
![Screenshot from 2021-03-28 18-07-12](https://user-images.githubusercontent.com/57143358/112752564-4a37e700-8ff1-11eb-9b06-87d0be4f3930.png)
![Screenshot from 2021-03-28 18-07-28](https://user-images.githubusercontent.com/57143358/112752565-4ad07d80-8ff1-11eb-9854-f2637e706744.png)

Alabhya268 commented:
@Ram81 ping for review

Alabhya268 commented:
> @Ram81, I've made execution time according to your requirements. Please have a look at the following images.
> 
> ![Screenshot from 2021-03-28 17-53-01](https://user-images.githubusercontent.com/57143358/112752562-486e2380-8ff1-11eb-96cf-a10fc8fc87e8.png)
> ![Screenshot from 2021-03-28 17-53-38](https://user-images.githubusercontent.com/57143358/112752563-499f5080-8ff1-11eb-8d48-7d58e9003ea4.png)
> ![Screenshot from 2021-03-28 18-07-12](https://user-images.githubusercontent.com/57143358/112752564-4a37e700-8ff1-11eb-9b06-87d0be4f3930.png)
> ![Screenshot from 2021-03-28 18-07-28](https://user-images.githubusercontent.com/57143358/112752565-4ad07d80-8ff1-11eb-9854-f2637e706744.png)

@Ram81 , Please have a look at this PR, I've made execution time according to your requirements.
